PR_REVIEW_CHECKLIST.md

```diff
@@ -0,0 +1,98 @@
+
+## Code patterns to check for improvement
+
+1. Is there appropriate error handling?
+2. Reuse...
+3. Are there any changes that could've been applied to one component that were applied to every usage of the component instead?
+    - Example: https://github.com/Expensify/App/pull/3129#pullrequestreview-668271068
+4. Ensure fixes for just android or ios aren't put in a `index.native.js`. Instead they should be in a specific `index.ios.js` or `index.android.js` file.
+    - Put another way, ensure platform-specific solutions are not applied cross-platform.
+5. Look for usages of react fragment wrappings and confirm they are necessary.
+    - Also look for any unnecessary `View`s and think if they should be replaced with React fragments.
+6. Were any new styles added? Can we reuse existing styles instead? Can we generalize the new style?
+
+## Code patterns to avoid
+
+1. Make sure any `setTimeout`s are absolutely necessary, not just a workaround for a situation that isn’t fully understood.
+2. Ensure Onyx is used appropriately. ([docs](https://github.com/expensify/react-native-onyx#merging-data))
+3. Don't mention the `isSmallScreenWidth` prop type explicitly in any component, use `...windowDimensionsPropTypes` instead.
+
+## Considerations around code reuse
+
+While trying to keep our code DRY (don't repeat yourself), you may find you're walking a thin line between refactoring & reusing a component vs. creating a brand new component. Here are some guidelines we use - they're not hard and fast rules, so please feel free to post in #expensify-open-source to gather peer opinions in order to make the best decisions.
+
+1. Specialization
+    - When one component is a special case of another, we should opt for the technique of composition. Here are the [React docs](https://reactjs.org/docs/composition-vs-inheritance.html#specialization) on Specialization. The idea is that it’s better to establish a pattern of creating increasingly specific components, instead of adding dozens of use cases to a single component.
+    - You might consider this when a component is being reused because the new use case is “close enough” to the original. Rather than adding functionality to that component with subtle additions, it may be better to create a new, more sprecialized version of that component.
+1. Refactors
+    - During a code review it becomes apparent that a refactor is needed. In this case, we recommend following these steps:
+        1. Identify the refactor and confirm with others that it’s needed.
+        2. Do that refactor first (as a separate job, if it qualifies), merge it, test for regressions.
+        3. Proceed with the original issue.
+
+# Submit / approve the PR
+
+While the steps above can be tedious to follow the first time you submit or review a PR, over time they will become more natural. The goal of these steps is to make sure we create the best product, with the best engineers writing the best code.
```

Comment:
Is it necessary to suggest that any part of this is tedious, unnatural, or difficult to follow? 

It's the job and this is what is required. If it is hard to follow then we are not doing a good job at communicating what needs to be done and should improve this document.

Reply:
"necessary"? Definitely not :D Maybe I like to explain things via [shit sandwhich](https://en.wiktionary.org/wiki/shit_sandwich) too much haha

Reply:
> If it is hard to follow then we are not doing a good job at communicating what needs to be done and should improve this document.

I don't think I was implying that this checklist is hard to follow, just that it can be time consuming the first time it's looked again. Again, I don't think it's necessary so I'm still happy to chop it

Reply:
Yeah, let's just assume that we are not asking for something that is even remotely painful (everyone will have their own opinion regardless). The idea behind the checklist is that it will make contributor jobs EASIER and not tedious.

Reply:
Good idea, I'll just remove this whole final section since it's definitely not important :D